### PR TITLE
[wip] Allow attribute selectors

### PR DESCRIPTION
**Pull request by nguyenj**
Don't pull this PR just yet, this is just to get the conversation started for providing [attribute selectors](https://github.com/kss-node/kss-node/issues/52).

I could use some help, it's not fully flushed out. Currently, there are:
- missing tests for this feature
- the markup output with the attributes are being escaped

### CODE REVIEW

**Comment by JohnAlbin** [comment]
Whoops. I just deleted the {{ markup }} Handlebars helper in favor of rendering the partial inside buildpage(); so you'll have to move that code over to the new spot.

**Comment by nguyenj** [comment]
@JohnAlbin I'm having a hard time figuring out the difference between twig and handlebars.

**Comment by JohnAlbin** [comment]
Feel free to ignore builder/base/twig and builder/twig for now. If you can get the Handlebars part working, I can port the functionality to the Twig builder. (The docs for Twig.js are crap.)

**Comment by nguyenj** [comment]
I don't have tests yet as I'm still trying to familiarize myself with the codebase; but I have most of the attributes working except that the attributes are being escaped when injected into the example markup. e.g. `[disabled="disabled"]` becomes `[disabled&#x3d;&quot;disabled&quot;]`

**Comment by JohnAlbin** [comment]
Handlebars allows you to skip the default HTML escaping if you use `{{{ var }}}` instead of the usual double-brackets.

**Comment by nguyenj** [comment]
I looked into [that](https://github.com/kss-node/kss-node/blob/master/builder/handlebars/index.hbs#L132-L134); but I don't think that's the issue. I'll take a look to see if it's being escaped when calling `toJSON()`.

**Comment by nguyenj** [comment]
Ah, you're right. I just have to pass in the `noEscape` option [here](https://github.com/kss-node/kss-node/blob/master/builder/base/handlebars/kss_builder_base_handlebars.js#L379).

**Comment by RobLoach** [comment]
This could likely do with a rebase.

**Comment by nguyenj** [comment]
@RobLoach I could use some help on this

File: builder/base/handlebars/kss_builder_base_handlebars.js [hunk 1]
```diff
@@ -427,7 +434,7 @@ class KssBuilderBaseHandlebars extends KssBuilderBase {
       if (section.markup) {
         // Load the information about this section's markup partial.
         let partialInfo = this.partials[section.reference];
-        let template = this.Handlebars.compile('{{> "' + partialInfo.name + '"}}');
+        let template = this.Handlebars.compile('{{> "' + partialInfo.name + '"}}', {noEscape: true});
```
Comment (RobLoach) [inline builder/base/handlebars/kss_builder_base_handlebars.js]: I believe John was refering to this one. Rather then `{{`, use the three braces.

**Comment by RobLoach** [comment]
This may also need the Twig support too.

**Comment by ryuran** [comment]
Some news about this PR ?
Need help ?

**Comment by nguyenj** [comment]
@ryuran help would be much appreciated

**Comment by nguyenj** [comment]
If you can help write some tests for these, that would be great.

**Comment by ryuran** [comment]
https://github.com/nguyenj/kss-node/pull/1 a start

**Comment by ryuran** [comment]
We should restart this work

**Comment by Didel** [comment]
@ryuran Yes please! I have also posted my ideas of the KSS-spec side of this in https://github.com/kneath/kss/issues/114 . Is there a specific thing others can help with?

**Pull request closed without merging**